packages/tasit-action/src/TasitAction.test.js

```diff
@@ -353,20 +398,27 @@ describe("TasitAction.Contract", () => {
         errorFn();
       };
 
+      txSubscription = simpleStorage.setValue("hello world");
+
       txSubscription.on("error", errorListener);
 
+      txSubscription.on("confirmation", confirmationListener);
+
+      await mineBlocks(provider, 5);
+
       const snapshotId = await createSnapshot(provider);
 
-      await mineBlocks(provider, 15);
+      await mineBlocks(provider, 20);
 
       expect(confirmationFn.called).to.be.true;
 
       await revertFromSnapshot(provider, snapshotId);
 
-      await mineBlocks(provider, 20);
+      // Note: Without that, ethers.provider will keep the same
+      // receipt.confirmations as before snapshot reversion
+      txSubscription.refreshProvider();
 
-      // Broadcast same transaction again (simulate reorg)
-      txSubscription = simpleStorage.setValue("hello world");
+      await mineBlocks(provider, 15);
```

Comment:
An idea that I think I've already mentioned elsewhere is that the "reorg" would be more clearly a reorg from reading the test code if you inserted another transaction along the way, like a including a fresh tx before the tx we're interested in and then reinserting the one of interest and then mining more blocks.

Reply:
But this can certainly come in a follow-up issue/PR that's not the highest priority

Reply:
I'm not sure if I've understood well.
Should I create some transaction before that 
```javascript
txSubscription = simpleStorage.setValue("hello world");
``` 
Is it?

Reply:
Yeah, I'd have to think carefully about whether this would work, but something like:

```
simpleStorage.setValue("hey earth");
await mineBlocks(provider, 1);
txSubscription = simpleStorage.setValue("hello world");
await mineBlocks(provider, 14);
```

Reply:
This can of course come in a later PR if you like that idea

Reply:
I've just committed and renamed the test case to a better name.

Reply:
Sorry, I've unchanged that because introduced a non-deterministic behavior, I'll create a new issue to do that carefully.

Reply:
Appended in already open issue https://github.com/tasitlabs/TasitSDK/issues/104

Reply:
Great - resolving this